packages/components/nodes/chatmodels/ChatFireworks/ChatFireworks.ts

```diff
@@ -0,0 +1,77 @@
+import { BaseCache } from '@langchain/core/caches'
+import { ChatFireworks } from '@langchain/community/chat_models/fireworks'
+import { ICommonObject, INode, INodeData, INodeOptionsValue, INodeParams } from '../../../src/Interface'
+import { getBaseClasses, getCredentialData, getCredentialParam } from '../../../src/utils'
+import { MODEL_TYPE, getModels } from '../../../src/modelLoader'
+
+class ChatFireworks_ChatModels implements INode {
+    label: string
+    name: string
+    version: number
+    type: string
+    icon: string
+    category: string
+    description: string
+    baseClasses: string[]
+    credential: INodeParams
+    inputs: INodeParams[]
+
+    constructor() {
+        this.label = 'ChatFireworks'
+        this.name = 'chatFireworks'
+        this.version = 1.0
+        this.type = 'ChatFireworks'
+        this.icon = 'Fireworks.png'
+        this.category = 'Chat Models'
+        this.description = 'Wrapper around Fireworks Chat Endpoints'
+        this.baseClasses = [this.type, ...getBaseClasses(ChatFireworks)]
+        this.credential = {
+            label: 'Connect Credential',
+            name: 'credential',
+            type: 'credential',
+            credentialNames: ['fireworksApi']
+        }
+        this.inputs = [
+            {
+                label: 'Cache',
+                name: 'cache',
+                type: 'BaseCache',
+                optional: true
+            },
+            {
```

Comment:
We need to have the model field parameter, example - https://github.com/FlowiseAI/Flowise/blob/main/packages/components/nodes/chatmodels/ChatHuggingFace/ChatHuggingFace.ts#L41

Since there are so many models, we can just make it as a `type: 'string` field, and get rid of the `listModels` method

Reply:
I modified that, not sure if I understand it correctly. Please let me know if other modification is needed.